File: java-example/src/main/java/org/onflow/examples/java/getNetworkParams/GetNetworkParametersAccessAPIConnector.java
package org.onflow.examples.java.getNetworkParams;

import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowChainId;

public class GetNetworkParametersAccessAPIConnector {
    private final FlowAccessApi accessAPI;

    public GetNetworkParametersAccessAPIConnector(FlowAccessApi accessAPI) {
        this.accessAPI = accessAPI;
    }

    public FlowChainId getNetworkParameters() {
        FlowAccessApi.AccessApiCallResponse<FlowChainId> response = accessAPI.getNetworkParameters();
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowChainId>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }
}

File: kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/getNetworkParams/GetNetworkParametersAccessAPIConnector.kt
package org.onflow.examples.kotlin.getNetworkParams

import org.onflow.flow.sdk.FlowAccessApi
import org.onflow.flow.sdk.FlowChainId

internal class GetNetworkParametersAccessAPIConnector(
    private val accessAPI: FlowAccessApi
) {
    fun getNetworkParameters(): FlowChainId {
        return when (val response = accessAPI.getNetworkParameters()) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }
}
